### ISSUE TITLE
Update the nginx.conf

### ISSUE THREAD

**Issue by mdartic** [seatable/seatable#36]
We encounter some problems due to the `nginx.conf`.

On the `http[s]://your_server_domain/workspace/:workspace_id/dtable/:dtable_name/` page, we had a blank page, and a 400 HTTP status code for the URL `http[s]://your_server_domain/dtable-server/dtables/:dtable_uuid?lang=fr`

We fix the issue by rewriting a `rewrite` nginx rule.

```
    location /dtable-server {
        # add_header Access-Control-Allow-Origin *;
        if ($request_method = 'OPTIONS') {
            add_header Access-Control-Allow-Origin *;
            add_header Access-Control-Allow-Methods GET,POST,PUT,DELETE,OPTIONS;
            add_header Access-Control-Allow-Headers "deviceType,token, authorization, content-type";
            return 204;
        }
        # rewrite /dtable-server/(.*) /$1 break;
        rewrite ^/dtable-server/(.*)$ /$1 break;
        proxy_pass         http://dtable_servers;
        proxy_redirect     off;
        proxy_set_header   Host              $host;
        proxy_set_header   X-Real-IP         $remote_addr;
        proxy_set_header   X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header   X-Forwarded-Host  $server_name;
        proxy_set_header   X-Forwarded-Proto $scheme;
    }
```

I can't do a PR, so I hope this will help you and you could add it to the next release, if you think it's fixing the rewrite rule.

**Comment by freeplant** [comment]
Why the original `rewrite /dtable-server/(.*) /$1 break;` can't work in your environment?

**Comment by mdartic** [comment]
I don't know exactly "why", but I can try to explain how it solved my issue.

Scenario / Step to reproduce :
* create a new machine that will host seatable
* add haproxy
* configure it to handle ssl and forward default path to the `backend_seatable`
```
# frontend part 
...
    use_backend backend_seatable if host_yourname
...
# backend part
backend backend_seatable
    http-request redirect scheme https code 301 if !{ ssl_fc }
    http-response set-header Strict-Transport-Security "max-age=16000000; preload;"
    server server1 127.0.0.1:8003
```
* install a fresh seatable with a URL like `http[s]://your_server_domain/` in environment
```
# docker-compose.yml, seatable part
    ports:
      - "8003:80"
    volumes:
      - seatable-data:/shared
      - ./conf/:/shared/seatable/conf/
    environment:
      - DB_HOST=db
      - DB_ROOT_PASSWD=***************
      - SEATABLE_SERVER_HOSTNAME=your_server_domain
      - TIME_ZONE=Europe/Paris
      - SEAFILE_ADMIN_EMAIL=*********
      - SEAFILE_ADMIN_PASSWORD=************
```
* update config files as said in #25
* start seatable
* create a superuser
* connect to seatable 
* create a new database
* click on the freshly created database
* get a blank page with a 400 error in console when the browser try to retrieve result from the URL `https://your_server_domain/dtable-server/dtables/:dtable_uuid?lang=fr`

I check first the logs (nginx mainly, but dtable too) and just see that I was getting 400 status. No log or errors in `dtable-sever.log` .

I read the nginx conf, and see that when we are doing an OPTIONS request, we have to receive a 204.
I try this one, and it's ok. A 204 response is well received.
But still 400 ones for others.

I decide to check if the nodejs process is really receiving request.
After reading the `package.json` of `/opt/seatable/seatable-server-latest/dtable-server` path, I make an npm ci to get all the dependencies of the project.
Then I change the `src/http-server.js` file and add some `logger.info` / `logger.error` at several point, and one in the `app.all("*"`. 
Then, I check again by making an `npm run build`, killing the actual node process, and relaunching it with a `node dist/src/index.js`.
I check the logs, but still no trace when I reload the seatable database page (in browser).

I copy the request made by my browser, and try to make this request with curl, but inside the docker.
By changing the URL, and use the "local one" (127.0.0.1:5000), it's a 200 response !

I think the problem comes from nginx of the seatable container.

I check the conf, see this rewrite rule after the 204, and change it for the new one `rewrite ^/dtable-server/(.*)$ /$1 break;`.
Restart nginx, make the `curl` in docker, still ok, and with the "external" URL (with `your_server_domain`), now a 200 !
Reload my browser, and, hourray !, now it's the database page :-) !

Maybe the fact we use HAProxy to handle SSL and redirect could introduce some changes in the nginx.conf ? Don't know exactly.
But, what is sure, it's the fact that the "external URL" is needed to be known by the underground layer of our server = SeaTable, dtable-server and so on.
The fact we have to change manually the `dtable_server_config.json` (related to #25), to allow HAProxy handle the SSL is not a good option (imho).
And, maybe, this introduce an error in the `nginx.conf` for SeaTable.

But, well, it's working !

Hope this will help you understand better.

And thank you by the way for all this work, seatable seems really great !

**Comment by freeplant** [comment]
Fixed in v0.9.8.

**Comment by mdartic** [comment]
Thanks !
